orangecontrib/timeseries/widgets/owlinechart.py

```diff
@@ -405,40 +411,47 @@ def setType(self, ax, type):
         ''' % locals())
 
 
-class LineChartContextHandler(DomainContextHandler):
+class LineChartContextHandler(ContextHandler):
     """
     Context handler of line chart. The specifics of this widget is
     that `attrs` variable is list of lists and it is not handled with
     the DomainContextHandler.
     """
+    def open_context(self, widget, domain_fetures):
+        if domain_fetures[0] is None:
+            # prevent opening context if no data
+            return
+        super().open_context(widget, domain_fetures)
 
-    def encode_setting(self, context, setting, value):
-        if setting.name == "attrs":
-            value = [
-                [(var.name, vartype(var)) for var in plot] for plot in value
-            ]
-        return super().encode_setting(context, setting, value)
-
-    def decode_setting(self, setting, value, domain=None, *_args):
-        decoded = super().decode_setting(setting, value, domain)
-        if setting.name == "attrs":
-            decoded = [[domain[name] for name, _ in plot] for plot in decoded]
-        return decoded
-
-    def match(self, context, _, attrs, metas):
-        if context.attributes == attrs and context.metas == metas:
-            return self.PERFECT_MATCH
+    def new_context(self, group):
+        context = super().new_context()
+        context.features = group[1]
+        return context
 
+    def match(self, context, domain_features, *args):
         if "attrs" not in context.values:
             return self.NO_MATCH
 
-        all_vars = attrs.copy()
-        all_vars.update(metas)
-
-        value = set(y for x in context.values["attrs"] for y in x)
-        return sum(all_vars.get(attr) == vtype for attr, vtype in value) / len(
-            value
-        )
+        domain, features = domain_features
+        attrs = context.values["attrs"]
+        if features is not None:
+            # when features are present match features only - find context with
+            # same features
+            if hasattr(context, "features") and context.features == features:
+                return self.PERFECT_MATCH
+            else:
+                return self.NO_MATCH
+        else:
+            # match if all selected attributes in domain
+            # when no features on input match just contexts with no features
+            values = set(y for x in attrs for y in x)
+            if (
+                context.features is None and len(values) > 0
```

Comment:
At this point the `context` may not have the `features` attribute.

One way of handling it, is to implement the `migrate_context`.

Reply:
Didn't think about that. Thank you. :) Now handler is even simpler.